### PR TITLE
cleanup  autoscale settings

### PR DESCRIPTION
**Pull request by rsethur**
# PR into Azure/azureml-examples

## Checklist

I have:

- [x] read and followed the contributing guidelines

## Changes

-

fixes #
cleanup autoscale settings created by online endpoint autoscale sample

### CODE REVIEW

**Review by petrodeg — APPROVED** [review]

File: cli/deploy-moe-autoscale.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@ DEPLOYMENT_RESOURCE_ID=$(az ml online-deployment show -e $ENDPOINT_NAME -n $DEPL
 # ARM id of the deployment. todo: change to --query "id"
 ENDPOINT_RESOURCE_ID=$(az ml online-endpoint show -n $ENDPOINT_NAME -o tsv --query "properties.\"azureml.onlineendpointid\"")
 # set a unique name for autoscale settings for this deployment. The below will append a random number to make the name unique.
-AUTOSCALE_SETTINGS_NAME=autoscale-`echo $RANDOM`
+AUTOSCALE_SETTINGS_NAME=autoscale-$ENDPOINT_NAME-$DEPLOYMENT_NAME-`echo $RANDOM`
```
Comment (magole) [inline cli/deploy-moe-autoscale.sh:24]: RANDOM will generate IDs from 0 to 32767. I recommend instead just computing a hash of string [WorkspaceName + EndpointName + DeploymentName] and then appending to the same string.

Comment (rsethur) [reply cli/deploy-moe-autoscale.sh:24]: thanks for the feedback prior the PR as well - helpful! For the samples i'm going to keep it simple.

Comment (magole) [reply cli/deploy-moe-autoscale.sh:24]: If you have workspace name available, prepend it to your string as well to minimize collisions.

File: cli/deploy-moe-autoscale.sh [hunk 2]
```diff
@@ -67,5 +67,9 @@ az monitor autoscale profile create \
 # </weekend_profile>
 
 # <delete_endpoint>
+# delete the autoscaling profile
+az monitor autoscale delete -n "$AUTOSCALE_SETTINGS_NAME"
+
+# delete the endpoint
 az ml online-endpoint delete --name $ENDPOINT_NAME --yes --no-wait
```
Comment (magole) [inline cli/deploy-moe-autoscale.sh:74]: Does it have to be "no wait"? Failures in deletion won't be visible.

File: cli/deploy-moe-autoscale.sh (hunk 1 above)
Comment (rsethur) [reply cli/deploy-moe-autoscale.sh:24]: noted.

File: cli/deploy-moe-autoscale.sh (hunk 2 above)
Comment (rsethur) [reply cli/deploy-moe-autoscale.sh:74]: we can remove --no-wait in main moe example.

**Pull request merged**